src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -76,16 +76,37 @@
         (rr/status ((response :outcome) outcomes->codes))
         (rr/content-type "text/plain"))))
 
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;;; Web app
-
 (defn try-to-parse
   [body]
   (try
     (cheshire/parse-stream (io/reader body) true)
     (catch Exception e
       (log/debug e))))
 
+(schema/defn handle-cert-clean
+  [{:keys [body]}
+   ca-settings :- ca/CaSettings]
+  (let [{:keys [certnames async]} (try-to-parse body)
+        [existing-certs missing-certs] (split-with
+                                        #(ca/certificate-exists? ca-settings %)
+                                        certnames)
+        message (when (seq missing-certs)
```

Comment:
Why does this need to be wrapped in `seq`?

Reply:
That's apparently preferred shorthand for `(not (empty?))` . See the note on [`empty?`](https://clojuredocs.org/clojure.core/empty_q).

Reply:
oh, interesting.